src/main/java/timetables_sort_use_case/frameworks_and_drivers/AllTimetablesScreen.java

```diff
@@ -58,33 +72,63 @@ public AllTimetablesScreen(JFrame frame, TimetablesSortController controller) {
      * opens TimeTableUI of the given timetable if a timetable button is clicked.
      */
     public void actionPerformed(ActionEvent e) {
-        System.out.println("Clicked " + e.getActionCommand());
-
         if (e.getActionCommand().equals("Sort")) {
             openTimetablesSortMenu();
-        } else if (e.getActionCommand().equals("mainMenu")) {
-
+        } else if (e.getActionCommand().equals("Main Menu")) {
+            openMainUI();
+        } else if (e.getActionCommand().equals("Overlap")) {
+            openOverlapUI();
         } else {
             int i = e.getActionCommand().length() - 1;
-            TimetableViewModel timetable = this.timetableViewModels[i];
-
+            openTimetableUI(i);
         }
     }
 
+    /**
+     * closes this view and opens a TimetablesSortMenu
+     */
     public void openTimetablesSortMenu() {
-        if (timetablesSortMenu != null) {
-            timetablesSortMenu.setVisible(false);
-            frame.remove(timetablesSortMenu);
-        }
-        timetablesSortMenu = new TimetablesSortMenu(this.frame, this);
+        timetablesSortMenu = new TimetablesSortMenu(frame, this);
+        frame.getContentPane().removeAll();
         frame.add(timetablesSortMenu);
-        this.setVisible(false);
-        timetablesSortMenu.setVisible(true);
-        this.frame.pack();
+        frame.pack();
     }
 
+    /**
+     * closes this view and opens MainUI
+     */
+    public void openMainUI() {
+        frame.getContentPane().removeAll();
+        frame.add(mainUI);
+        frame.revalidate();
+    }
+
+    /**
+     * closes this view and opens TimeTableUI with the timetable that the user chose
+     *
+     * @param i the index of the timetable that was chosen
+     */
+    public void openTimetableUI(int i) {
+        frame.getContentPane().removeAll();
+        allTimetablesController.updateSubscribers(i);
+        timetableUI.updateTimetable();
+        frame.add(timetableUI);
+        frame.revalidate();
+    }
+
+    /**
+     * closes this view and opens OverlapInputDialogue
+     */
+    public void openOverlapUI() {
+        overlapInputDialog.setVisible(true);
```

Comment:
Um, this doesn't seem to close this view. Is the comment wrong?

Reply:
comment is wrong yes, i fogot that your view is a dialogue, i fixed it